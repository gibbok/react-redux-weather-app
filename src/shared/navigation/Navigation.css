.activeLink{
    color:green;
    font-weight: bold;
}