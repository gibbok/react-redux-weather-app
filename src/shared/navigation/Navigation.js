// @flow
import * as React from 'react'
import { NavLink } from 'react-router-dom'

/* eslint-disable no-undef */
type PropsType = {
  +onClick: Function,
  +id: number,
  +title:string,
  +tooltip:string,
  +url:string
}
/* eslint-enable no-undef */

const Navigation:React.StatelessFunctionalComponent<any> = ({ onClick, id, title, tooltip, url }:PropsType):React.Element<any> => (
  <div onClick={onClick} alt={tooltip}>
    <NavLink
      to={url}
      exact
      activeStyle={{
        fontWeight: 'bold',
        color: 'red'
      }}
    >{title}</NavLink >
  </div >
)

export default Navigation
