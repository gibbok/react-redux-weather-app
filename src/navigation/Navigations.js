import React from 'react'
import Navigation from './Navigation'

const NavigationList = ({ navigations, onNavigationClick }) => (
  <ul>
    {navigations.map((navigation, index) =>
      <Navigation
        key={index}
        {...navigations[index]}
        onClick={() => onNavigationClick(navigation)}
      />
    )}
  </ul>
)

export default NavigationList
