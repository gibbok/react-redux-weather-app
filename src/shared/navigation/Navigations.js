// @flow
import * as React from 'react'
import Navigation from './Navigation'
import { Drawer, List, withStyles } from 'material-ui'
// import HomeIcon from 'material-ui-icons/Home'
// import MapIcon from 'material-ui-icons/Map'

/* eslint-disable no-undef */
type PropsType = {
  +navigations: Array<{
    +id:string,
    +isActive:boolean,
    +title:string,
    +tooltip:string,
    +url:string,
  }>,
  +onNavigationClick: Function,
  +classes: Object
}
/* eslint-enable no-undef */
const drawerWidth = 240
const styles = theme => ({
  drawerPaper: {
    position: 'relative',
    height: '100%',
    width: drawerWidth
  },
  drawerHeader: theme.mixins.toolbar
})

const Navigations:React.StatelessFunctionalComponent<any> = ({ navigations, onNavigationClick, classes }:PropsType):React.Element<any> => {
  return (
    <div>
      <Drawer
        type='permanent'
        classes={{
          paper: classes.drawerPaper
        }}
        >
        <List>
          {navigations.map((navigation, index) =>
            <Navigation
              key={index}
              {...navigations[index]}
              onClick={() => onNavigationClick(navigation)}
            />
           )}
        </List>
      </Drawer>
    </div>
  )
}

export default withStyles(styles)(Navigations)
