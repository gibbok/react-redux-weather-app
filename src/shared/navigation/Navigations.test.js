import React from 'react'
import {shallow} from 'enzyme'
import Navigations from './Navigations'
import Navigation from './Navigation'

describe('<Navigations />', () => {
  it('should render two <Navigation /> components', () => {
    const data = [
      {
        id: 'home',
        title: 'Weather',
        tooltip: 'Weather',
        isActive: true,
        url: '/'
      },
      {
        id: 'map',
        title: 'Map',
        tooltip: 'Weather map',
        isActive: false,
        url: '/map'
      }
    ]
    const wrapper = shallow(
      <Navigations
        navigations={data}
      />
    )
    expect(wrapper.find(Navigation).length).toBe(2)
  })
})
