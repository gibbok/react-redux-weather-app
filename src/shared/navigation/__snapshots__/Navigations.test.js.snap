// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigations /> should render 1`] = `
<div>
  <withStyles(Drawer)
    classes={
      Object {
        "paper": "Navigations-drawerPaper-1",
      }
    }
    type="permanent"
  >
    <div
      className="Navigations-drawerInner-3"
    >
      <div
        className="Navigations-drawerHeader-4"
      >
        <withStyles(IconButton)>
          <pure(ChevronLeft) />
        </withStyles(IconButton)>
      </div>
      <withStyles(Divider) />
      <withStyles(List)>
        <withStyles(Navigation)
          id="home"
          isActive={true}
          onClick={[Function]}
          title="Weather"
          tooltip="Weather"
          url="/"
        />
        <withStyles(Navigation)
          id="map"
          isActive={false}
          onClick={[Function]}
          title="Map"
          tooltip="Weather map"
          url="/map"
        />
      </withStyles(List)>
    </div>
  </withStyles(Drawer)>
</div>
`;
